src/helpers.cpp

```diff
@@ -246,6 +246,29 @@ namespace ada::helpers {
     if (url.query.has_value()) return;
     while (!url.path.empty() && url.path.back() == ' ') { url.path.resize(url.path.size()-1); }
   }
+
+  ada_really_inline size_t find_authority_delimiter(const bool& is_special, std::string_view view) noexcept {
+    size_t location = 0;
+    bool location_found{false};
```

Comment:
Do you need location_found? It seems to me that if you exit the loop with location_found<view.size(), then location_found is true, otherwise it is false. So there, you are done.

(Note that I don't expect that this code simplification would change anything at all the performance, but I think we can agree than having fewer variables is better.)

Reply:
Can be simplified with `std::min(location, view.size());`